src/openforms/authentication/contrib/digid_oidc/tests/test_auth_procedure.py

```diff
@@ -0,0 +1,277 @@
+from urllib.parse import parse_qs, urlparse
+
+from django.test import TestCase, override_settings
+from django.urls import reverse
+
+import requests_mock
+from rest_framework import status
+
+from openforms.forms.tests.factories import (
+    FormDefinitionFactory,
+    FormFactory,
+    FormStepFactory,
+)
+
+from ..models import OpenIDConnectPublicConfig
+
+
+@override_settings(CORS_ALLOW_ALL_ORIGINS=True, IS_HTTPS=True)
+class DigiDOIDCTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        config = OpenIDConnectPublicConfig.get_solo()
+        config.enabled = True
+        config.oidc_rp_client_id = "testclient"
+        config.oidc_rp_client_secret = "secret"
+        config.oidc_rp_sign_algo = "RS256"
+        config.oidc_rp_scopes_list = ["openid", "bsn"]
+        config.oidc_redirect_allowed_hosts = []
+
+        config.oidc_op_jwks_endpoint = (
+            "http://identityprovider/auth/realms/master/protocol/openid-connect/certs"
+        )
+        config.oidc_op_authorization_endpoint = (
+            "http://identityprovider/auth/realms/master/protocol/openid-connect/auth"
+        )
+        config.oidc_op_token_endpoint = (
+            "http://identityprovider/auth/realms/master/protocol/openid-connect/token"
+        )
+        config.oidc_op_user_endpoint = "http://identityprovider/auth/realms/master/protocol/openid-connect/userinfo"
+        config.save()
+
+    def test_redirect_to_digid_oidc(self):
+        form = FormFactory.create(authentication_backends=["digid_oidc"])
+        form_definition = FormDefinitionFactory.create(login_required=True)
+        FormStepFactory.create(form_definition=form_definition, form=form)
```

Comment:
you can get a full form configured through the `FormFactory` - check for usages of `FormFactory.create(generate_minimal_config=...)`